test/pet.test.js

```diff
@@ -0,0 +1,189 @@
+const Pet = require('../src/pet');
+
+let testPet;
+
+const deadMessage = "Your pet is no longer alive :("
+
+beforeEach(() => {
+    testPet = new Pet();
+  });
+
+describe('constructor', () => {
+    it('should return a new pet with a name', () => {
+        expect(new Pet()).toBeInstanceOf(Object);
+        expect(new Pet().name).toBe("An unloved pet");
+        expect(new Pet('Manny').name).toBe("Manny");
+    })
+
+    it('should return a new pet with an age', () => {
+        expect(new Pet().age).toBe(0);
+    })
+
+    it('shold return a new pet with a hunger', () => {
+        expect(new Pet().hunger).toBe(0);
+    })
+
+    it('shold return a new pet with a fitness', () => {
+        expect(new Pet().fitness).toBe(10);
+    })
+})
+
+describe('growUp', () => {
+    it('should make the pet grow', () => {
+        testPet.growUp();
+        expect(testPet.age).toBe(1);
+    })
+
+    it('should make the pet hunger increase', () => {
+        testPet.growUp();
+        expect(testPet.hunger).toBe(5);
+    })
+
+    it('should make the pet fitness decrease', () => {
+        testPet.growUp();
+        expect(testPet.fitness).toBe(7);
+    })
+
+    it('should throw an exception with "Your pet is no longer alive :(", if dead', () => {
+        testPet.age = 30;
+        expect(() => testPet.growUp()).toThrow(deadMessage);
+    })
+})
+
+describe('walk', () => {
+    it('should increase fitness by 4', () => {
+        testPet.fitness = 5;
+        testPet.walk();
+        expect(testPet.fitness).toBe(9);
+    })
+
+    it('should have a max fitness of 10', () => {
+        testPet.walk();
```

Comment:
instead of relying on  previous function calls in other tests, it would be better to define what the testPet.fitness is that you are testing against in each it bloc, so that if you ever come back to this code again and only look at one function, you know exactly what it is doing, right there. Otherwise you may have to go through the whole suite just to find out why 1 test was not working.

Reply:
yup! Added that in now 👍